extensions/charts/rke2template.go

```diff
@@ -0,0 +1,122 @@
+package charts
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/rancher/shepherd/clients/rancher"
+	"github.com/rancher/shepherd/extensions/cloudcredentials"
+	"github.com/rancher/shepherd/extensions/clusters/kubernetesversions"
+	"github.com/rancher/shepherd/extensions/defaults"
+	"github.com/rancher/shepherd/pkg/api/steve/catalog/types"
+	"github.com/rancher/shepherd/pkg/namegenerator"
+	"github.com/rancher/shepherd/pkg/wait"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/watch"
+)
+
+const (
+	fleetNamespace = "fleet-default"
+)
+
+// InstallAWSOutOfTreeChart installs the CSI chart for aws cloud provider in a given cluster.
+func InstallRKE2TemplateChart(client *rancher.Client, installOptions *InstallOptions, repoName, templateName string, credentials *cloudcredentials.CloudCredential) error {
+	serverSetting, err := client.Management.Setting.ByID(serverURLSettingID)
+	if err != nil {
+		return err
+	}
+
+	registrySetting, err := client.Management.Setting.ByID(defaultRegistrySettingID)
+	if err != nil {
+		return err
+	}
+
+	chartInstallActionPayload := &payloadOpts{
+		InstallOptions:  *installOptions,
+		Name:            templateName,
+		Namespace:       fleetNamespace,
+		Host:            serverSetting.Value,
+		DefaultRegistry: registrySetting.Value,
+	}
+
+	k8sversions, err := kubernetesversions.Default(client, "rke2", nil)
+	if err != nil {
+		return err
+	}
+
+	chartValues, err := client.Catalog.GetChartValues(repoName, templateName, installOptions.Version)
+	if err != nil {
+		return err
+	}
+
+	chartInstallAction := RKE2TemplateInstallAction(chartInstallActionPayload, repoName, templateName, credentials.ID, k8sversions[0], fleetNamespace, chartValues)
+
+	catalogClient, err := client.GetClusterCatalogClient(installOptions.Cluster.ID)
+	if err != nil {
+		return err
+	}
+
+	err = client.Catalog.InstallChart(chartInstallAction, repoName)
+	if err != nil {
+		return err
+	}
+
+	client.Session.RegisterCleanupFunc(func() error {
+		err := client.Catalog.UninstallChart(templateName, fleetNamespace, newChartUninstallAction())
+		if err != nil {
+			return err
+		}
+
+		watchAppInterface, err := catalogClient.Apps(fleetNamespace).Watch(context.TODO(), metav1.ListOptions{
+			FieldSelector:  "metadata.name=" + templateName,
+			TimeoutSeconds: &defaults.WatchTimeoutSeconds,
+		})
+		if err != nil {
+			return err
+		}
+
+		err = wait.WatchWait(watchAppInterface, func(event watch.Event) (ready bool, err error) {
+			if event.Type == watch.Error {
+				return false, fmt.Errorf("there was an error uninstalling rke2 template chart")
```

Comment:
Sam since this is an Errorf, can you add the error return as part of the message?